creator-node/src/services/SessionExpirationQueue.js

```diff
@@ -0,0 +1,110 @@
+const Bull = require('bull')
+const Sequelize = require('sequelize')
+const sessionManager = require('../sessionManager')
+const config = require('../config')
+const { logger } = require('../logging')
+const { SessionToken } = require('../models')
+
+
+const QUEUE_INTERVAL_MS = 60 * 1000 * 60 * 24 // daily run
+const SESSION_EXPIRATION_AGE = 60 * 1000 * 60 * 24 * 14 // 2 weeks 
+
+const PROCESS_NAMES = Object.freeze({
+  expire_sessions: 'expire_sessions'
+})
+
+/**
+ * A persistent cron-style queue that periodically deletes expired session tokens from Redis cache and the database.
+ *
+ * The queue runs daily on cron.
+ * 
+ */
+class SessionExpirationQueue {
+  constructor () {
+    this.queue = new Bull(
+      'session-expiration-queue',
+      {
+        redis: {
+          port: config.get('redisPort'),
+          host: config.get('redisHost')
+        },
+        defaultJobOptions: {
+          removeOnComplete: true,
+          removeOnFail: true
+        }
+      }
+    )
+
+    // Clean up anything that might be still stuck in the queue on restart
+    this.queue.empty()
+
+    this.queue.process(
+      PROCESS_NAMES.expire_sessions,
+      /* concurrency */ 1,
+      async (job, done) => {
+        try {
+          this.logStatus('Starting')
+          let progress = 0
+
+          const SESSION_EXPIRED_CONDITION = {
+            where: {
+              created_at: {
+                [Op.gt]: new Date(Date.now() - SESSION_EXPIRATION_AGE)
+              }
+            }
+          }
+          const BATCH_COUNT = 100
+
+          const numExpiredSessions = await SessionToken.count(SESSION_EXPIRED_CONDITION)
+          this.logStatus(`${numExpiredSessions} expired sessions ready for deletion.`) // TODO refactor for readability
+
+          for (let numRemainingSessionsToExpire = numExpiredSessions; numRemainingSessionsToExpire >= BATCH_COUNT; numRemainingSessionsToExpire -= BATCH_COUNT) {
+            const sessionsToDelete = await SessionToken.findAll(Object.assign({}, SESSION_EXPIRED_CONDITION, { limit: BATCH_COUNT }))
+            sessionsToDelete.forEach(({ token }) => {
+              await sessionManager.deleteSession(token) // TODO is it more efficient to delete each session one-by-one in Redis (bulk delete not possible in Redis) and the DB, or to rely on Redis for delete and DBManager.deleteSessionTokensFromDB for bulk deletion on the DB side?
```

Comment:
ok to do sessionManager delete, but you can do multiexec in redis - like a transaction block in postgres, so you can batch delete ops that way - can save a bit of latency.